kani-driver/src/concrete_playback.rs

```diff
@@ -12,12 +12,84 @@ use concrete_vals_extractor::{extract_harness_values, ConcreteVal};
 use kani_metadata::HarnessMetadata;
 use std::collections::hash_map::DefaultHasher;
 use std::ffi::OsString;
-use std::fs::{self, File};
+use std::fs::File;
 use std::hash::{Hash, Hasher};
-use std::io::{Read, Write};
+use std::io::{BufRead, BufReader, Write};
 use std::path::Path;
 use std::process::Command;
 
+use self::tempfile::TempFile;
+
+/// Handle a tempfile functionality
+/// for example - used during concrete playback's inplace run to write to the source file
+mod tempfile {
+    use std::{
+        env,
+        fs::{self, rename, File},
+        io::{BufWriter, Error, Write},
+        path::PathBuf,
+    };
+
+    use crate::util;
+
+    pub struct TempFile {
+        pub file: File,
+        pub temp_path: PathBuf,
+        pub writer: Option<BufWriter<File>>,
+        renamed: bool,
+    }
+
+    impl TempFile {
+        /// Create a temp file
+        pub fn try_new() -> Result<Self, Error> {
+            let mut temp_path = env::temp_dir();
```

Comment:
You need to generate a somewhat random name to avoid collision. See the [`temp_dir` documentation](https://doc.rust-lang.org/std/env/fn.temp_dir.html):

> thus, the creation of any files or directories in the temporary directory must use a secure method to create a uniquely named file.